mantidproject/mantid
# Support save and load of plots containing axlines

This PR adds support for saving and loading plots containing `axvline`s and `axhline`s. This is been achieved by adding them to the `creation_args` of a `mantidaxes`, and replaying them on load. 



**To test:**

1. Run the following script
```
from mantid.simpleapi import *
import matplotlib.pyplot as plt

ws1 = CreateSampleWorkspace(OutputWorkspace="ws1")
fig, axes = plt.subplots(edgecolor='#ffffff', subplot_kw={'projection': 'mantid'})
axes.plot(ws1, color='#ff7f0e', label='ws1', specNum=2)
axes.axhline(y=1, xmin=0, xmax=1)
axes.axvline(2500, 0, 1)
plt.show()
```
2. With the plot open, save the project.
3. Close the plot and open the project you just saved.
4. The plot should reappear as before. All lines should be present and there should be no errors.

Fixes #28985 
Fixes #29349 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
